### PR TITLE
Require Python >=3.5

### PR DESCRIPTION
**Pull request by minrk**
Pythons 2.7-3.5 are EOL, but 3.5 is still the default python3 in a supported Ubuntu LTS for one more year

### CODE REVIEW

**Comment by minrk** [comment]
Everything's okay except for pypy3. Going to keep digging...

**Comment by minrk** [comment]
Success!

File: setup.py [hunk 1]
```diff
@@ -1191,64 +1197,70 @@ class CythonCommand(build_ext_c):
...
         """hack to build pypy extension only after building bundled libzmq
 
         otherwise it will fail when libzmq is bundled.
         """
-        def build_extensions(self):
-            self.extensions.remove(dummy_ext)
-            bld_ext.build_extensions(self)
+
+        def finalize_options(self):
+            super().finalize_options()
```
Comment (bluca) [inline setup.py:1252]: @minrk this gives me an error when building on Debian testing:

```
[  400s] running build_ext
[  400s] Traceback (most recent call last):
[  400s]   File "setup.py", line 1374, in <module>
[  400s]     setup(**setup_args)
[  400s]   File "/usr/lib/pypy/dist-packages/setuptools/__init__.py", line 162, in setup
[  400s]     return distutils.core.setup(**attrs)
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/core.py", line 151, in setup
[  400s]     dist.run_commands()
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 953, in run_commands
[  400s]     self.run_command(cmd)
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 972, in run_command
[  400s]     cmd_obj.run()
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/command/build.py", line 127, in run
[  400s]     self.run_command(cmd_name)
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/cmd.py", line 334, in run_command
[  400s]     self.distribution.run_command(command)
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 971, in run_command
[  400s]     cmd_obj.ensure_finalized()
[  400s]   File "/usr/lib/pypy/lib-python/2.7/distutils/cmd.py", line 109, in ensure_finalized
[  400s]     self.finalize_options()
[  400s]   File "setup.py", line 1252, in finalize_options
[  400s]     super().finalize_options()
[  400s] TypeError: __init__() takes at least 2 arguments (1 given)
[  400s] E: pybuild pybuild:352: build: plugin distutils failed with: exit code=1: /usr/bin/pypy setup.py build 
```

Any idea?

Comment (minrk) [reply setup.py:1252]: yes, that's Python 2.7 which is no longer supported.

Comment (bluca) [reply setup.py:1252]: Uh weird, I removed all the python2 stuff. Maybe it's the shbang in the setup.py?

Comment (bluca) [reply setup.py:1252]: Ah it was pypy - didn't notice there's a separate pypy3 package to use

**Pull request merged**